src/java/org/apache/cassandra/db/guardrails/GuardrailsConfig.java

```diff
@@ -49,34 +51,72 @@
     boolean getEnabled();
 
     /**
-     * @return The threshold to warn or abort when creating more tables than threshold.
+     * @return The threshold to warn or abort when creating more user keyspaces than threshold.
+     */
+    IntThreshold getKeyspaces();
+
+    /**
+     * @return The threshold to warn or abort when creating more user tables than threshold.
      */
-    Threshold.Config getTables();
+    IntThreshold getTables();
```

Comment:
Returning our interface we are decoupling config from the actual guardrails (`Threshold` and `Values`).